pkg/admission/decode.go

```diff
@@ -53,7 +57,18 @@ func decodeInfrastructureConfig(decoder runtime.Decoder, infra *core.ProviderCon
 	return infraConfig, nil
 }
 
-func decodeCloudProfileConfig(decoder runtime.Decoder, config *gardencorev1beta1.ProviderConfig) (*gcp.CloudProfileConfig, error) {
+// DecodeCloudProfileConfig decodes the `CloudProfileConfig` from the given `ProviderConfig`.
+func DecodeCloudProfileConfig(decoder runtime.Decoder, config *core.ProviderConfig) (*gcp.CloudProfileConfig, error) {
+	cloudProfileConfig := &gcp.CloudProfileConfig{}
+	if err := util.Decode(decoder, config.Raw, cloudProfileConfig); err != nil {
+		return nil, err
+	}
+
+	return cloudProfileConfig, nil
+}
+
+// DecodeCloudProfileFromExternalConfig decodes the `CloudProfileConfig` from the given `ProviderConfig`.
+func DecodeCloudProfileFromExternalConfig(decoder runtime.Decoder, config *gardencorev1beta1.ProviderConfig) (*gcp.CloudProfileConfig, error) {
```

Comment:
Sorry for being so nit-picking, but do you want to rename this to `DecodeCloudProfileConfigFromExternalProviderConfig` or something similar? Cause this is rather decoding the provider config than the actual `CloudProfile`.
```suggestion
func DecodeCloudProfileConfigFromExternalProviderConfig(decoder runtime.Decoder, config *gardencorev1beta1.ProviderConfig) (*gcp.CloudProfileConfig, error) {
```